### PR TITLE
Modify the return value of the parseLeaderValV3 interface

### PR DESCRIPTION
**Pull request by monadbobo**

### CODE REVIEW

**Review by panda-sheep — APPROVED** [review]
Good

**Comment by critical27** [comment]
Something go wrong here, ut failed.

**Comment by monadbobo** [comment]
> Something go wrong here, ut failed.
Already fixed.

File: src/meta/processors/jobMan/MetaJobExecutor.cpp [hunk 1]
```diff
@@ -117,20 +117,18 @@ ErrOrHosts MetaJobExecutor::getLeaderHost(GraphSpaceID space) {
...
+        auto result = MetaServiceUtils::parseLeaderValV3(leaderIter->val());
+        if (!ok(result)) {
             continue;
         }
-        auto it = std::find_if(hosts.begin(), hosts.end(), [&](auto& item){
-            return item.first == host;
-        });
+        auto [host, _] = value(result);
+        auto it = std::find_if(
+            hosts.begin(), hosts.end(), [host = host](auto& item) { return item.first == host; });
```
Comment (bright-starry-sky) [inline src/meta/processors/jobMan/MetaJobExecutor.cpp:129]: Redundant copy for param `host`.

File: src/meta/processors/partsMan/ListHostsProcessor.cpp [hunk 2]
```diff
@@ -190,19 +187,21 @@ nebula::cpp2::ErrorCode ListHostsProcessor::fillLeaders() {
...
             continue;  // skip inactive host
         }
 
-        auto hostIt = std::find_if(hostItems_.begin(), hostItems_.end(), [&](const auto& hit) {
-            return hit.get_hostAddr() == host;
-        });
+        auto hostIt = std::find_if(
+            hostItems_.begin(), hostItems_.end(), [host = std::move(host)](const auto& hit) {
+                return hit.get_hostAddr() == host;
+            });
```
Comment (bright-starry-sky) [inline src/meta/processors/partsMan/ListHostsProcessor.cpp:204]: `std::find` is enough.

**Review by bright-starry-sky — COMMENTED** [review]
Good job. LGTM.

**Review by critical27 — APPROVED** [review]

**Review by panda-sheep — APPROVED** [review]

File: src/meta/processors/jobMan/MetaJobExecutor.cpp (hunk 1 above)
Comment (monadbobo) [reply src/meta/processors/jobMan/MetaJobExecutor.cpp:129]: If you don't copy here, it will cause a clang compilation error (possibly a compiler bug).

**Pull request closed without merging**